server/src/internalClusterTest/java/org/elasticsearch/repositories/SnapshotMetricsIT.java

```diff
@@ -132,7 +135,7 @@ public void testSnapshotAPMMetrics() throws Exception {
         }
 
         // wait for snapshot to finish to test the other metrics
-        awaitNumberOfSnapshotsInProgress(0);
+        safeGet(snapshotFuture);
```

Comment:
I think if we wait for the response to be received that should mean the metrics should be written?

I'm running in a loop to try and confirm that now

Reply:
Another source of flakiness is the tension between the throttling and `safeGet`. If the setting was too high, we exceeded the `safeGet` timeout, if the setting was too low, we didn't see any throttling.

For this reason I've [moved out the test for the throttling metrics to another test](https://github.com/elastic/elasticsearch/pull/132780/commits/f034e1d3a4b4923c521e8a8a2448370f97acf82e). It has more complexity to ensure we see throttling and we don't run too long.